crates/ruma-events/src/call/member.rs

```diff
@@ -90,20 +90,22 @@ pub struct Membership {
 impl Membership {
     /// Application is "m.call" and scope is "m.room"
     pub fn is_room_call(&self) -> bool {
-        if let Application::Call(call) = &self.application {
-            return call.scope == CallScope::Room;
-        }
-        false
+        self.call_content().and_then(call.scope == CallScope::Room).unwrap_or(false)
     }
 
     /// Application is "m.call"
     pub fn is_call(&self) -> bool {
+        self.call_content().is_some()
+    }
+
+    /// Get a reference to the call content if the application is "m.call"
+    /// otherwise it returns None.
+    pub fn call_content(&self) -> Option<&CallApplicationContent> {
         match self.application {
-            Application::Call(_) => true,
-            _ => false,
+            Application::Call(call) => Some(&call),
+            _ => None,
         }
```

Comment:
We have a nice macro for this: You can replace the whole body of this function with `as_variant!(self, Application::Call)` (`use as_variant::as_variant;`)

Reply:
this is super helpful!